test/integration/failover_stateful_test.lua

```diff
@@ -235,6 +239,86 @@ function g.test_leader_restart()
     end)
 end
 
+function g.test_leader_promote()
+    g.kingdom:connect_net_box()
+    helpers.retrying({}, function()
+        g.kingdom.net_box:call('get_coordinator')
+    end)
+
+    -------------------------------------------------------
+
+    local storage = g.cluster:server('storage-1')
+    local ok, err = storage.net_box:eval([[
+        return require('cartridge').failover_promote(...)
+    ]], {{[storage_uuid] = storage_2_uuid}})
+
+    t.assert(ok, err ~= nil and err.err)
+    t.assert_equals(err, nil)
+
+    helpers.retrying({}, function()
+        t.assert_equals(eval('router',    q_leadership), storage_2_uuid)
+        t.assert_equals(eval('storage-1', q_leadership), storage_2_uuid)
+        t.assert_equals(eval('storage-2', q_leadership), storage_2_uuid)
+        t.assert_equals(eval('storage-3', q_leadership), storage_2_uuid)
+    end)
+
+    local ok, err = storage.net_box:eval([[
+        local cartridge = require('cartridge')
+        return cartridge.failover_promote(...)
+    ]], {{[storage_uuid] = storage_1_uuid}})
+
+    t.assert(ok, err ~= nil and err.err)
+    t.assert_equals(err, nil)
+
+    helpers.retrying({}, function()
+        t.assert_equals(eval('router',    q_leadership), storage_1_uuid)
+        t.assert_equals(eval('storage-1', q_leadership), storage_1_uuid)
+        t.assert_equals(eval('storage-2', q_leadership), storage_1_uuid)
+        t.assert_equals(eval('storage-3', q_leadership), storage_1_uuid)
+    end)
+
+    -------------------------------------------------------
+
+    local ok, err = storage.net_box:eval([[
+        local cartridge = require('cartridge')
+        return cartridge.failover_promote(...)
+    ]], {{[storage_uuid] = 'invalid_uuid'}})
+
+    t.assert_equals(ok, nil)
+    t.assert_not_equals(err, nil)
```

Comment:
Check class_name and message with `assert_covers`

Reply:
Fixed